### PR TITLE
New IA idea for GetTor (tips to download the Tor Browser and get bridges)

### PR DESCRIPTION
**Pull request by ilv**
**What does your Instant Answer do?**
It gives useful tips to get the Tor Browser and bridges, especially for users that don't have access to Tor Project's official website (TPO).

**What problem does your Instant Answer solve (Why is it better than organic links)?**
If you can't access TPO (i.e. it's blocked), then is probably that you won't know these tips. DDG will provide an alternative source of information for people looking to circumvent censorship.

**What are some example queries that trigger this Instant Answer?**
Example 1: get tor
Example 2: get tor browser
Example 3: get bridges
Example 4: get tor bridges

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
People that live and/or work in highly censored places.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
It's not.

**Which existing Instant Answers will this one supercede/overlap with?**
None that I'm aware of.

**Are you having any problems? Do you need our help with anything?**
I'm not sure if the amount of text displayed is reasonable. What do you think?

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
ddg-gettor

ddg-getbridges

## Checklist
Please place an 'X' where appropriate.

```
- [X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [X] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - So far I've only tested it on the Virtual Box virtual machine provided for development.

### CODE REVIEW

**Comment by ilv** [comment]
Some links about the current approaches to deliver the Tor Browser and bridges:

1) [GetTor](https://bridges.torproject.org/)
2) [BridgeDB](https://bridges.torproject.org/)

**Comment by moollaza** [comment]
@ilv thanks a lot for contributing, we really appreciate it! We'll be sure to review this shortly and give you some feedback!

**Comment by ilv** [comment]
@moollaza Thanks to you people! I look forward to your feedback!

**Comment by isislovecruft** [comment]
@ilv Perhaps it would be better to direct users to the recommended pages at https://bridges.torproject.org and https://www.torproject.org/download/download-easy.html first, rather than telling them about the failsafe distribution mechanisms as the "default" answer? Or perhaps we should tell them about the email options if the user's query was more like "tor browser website not working"?

**Comment by ilv** [comment]
@isislovecruft Thanks for the feedback. I agree on adding a reference to the recommended pages (will work on it during the week). I do think we should keep the same triggers, **and** add more triggers of the form "tor browser website not working", "bridge website is blocket", etc.

**Comment by moollaza** [comment]
@ilv terribly sorry for the delay. I'll try and get around to this soon!

File: lib/DDG/Goodie/GetTor.pm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+triggers any => 'gettor',
+                'get tor',
+                'gettor browser',
+                'gettor bridges',
+                'get bridges',
+                'tor bridges',
+                'tor browser',
+                'torbrowser';
+
+handle query_lc => sub {
```
Comment (moollaza) [inline lib/DDG/Goodie/GetTor.pm]: Goodie templates have now been documented. Let's switch this over to use the `text` template!

https://duck.co/duckduckhack/goodie_displaying

Comment (moollaza) [inline lib/DDG/Goodie/GetTor.pm]: Let's verify the query fits our expected structure using a regex, or at the very least, lets make a check that the remiander of the query doesn't contains things we don't want.

The regex approach is the stronges:

```perl
return unless m/(?:how can I|how to)?(?:get ?|download ) tor(?: bridges|browser)?/;
```

Something like that^ should work. I haven't tested it, just writing that on-the-fly!

**Comment by ilv** [comment]
@moollaza no worries. I'll try to take a look at your notes soon, probably next week.

**Comment by mintsoft** [comment]
Already merged in #1040

**Pull request closed without merging**